File: 100-999/104. Maximum Depth of Binary Tree/solution_1.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/**
 * @param {TreeNode} root
 * @return {number}
 */

var maxDepth = function(root) 
{
    let depth = 0
    
    if (root)
    {
        let nodes = [root]

        while (nodes.length > 0)
        {
            let nextNodes = []

            for (const node of nodes) 
            {
                if (node.left)
                    nextNodes.push(node.left)
                
                if (node.right)
                    nextNodes.push(node.right)
            }

            nodes = nextNodes

            depth++;
        }
    }

    return depth
};

File: 100-999/104. Maximum Depth of Binary Tree/solution_1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution
{
    public int MaxDepth(TreeNode root)
    {
        int depth = 0;

        if (root != null)
        {
            List<TreeNode> nodes = new List<TreeNode> { root };

            while (nodes.Count > 0)
            {
                List<TreeNode> nextNodes = new List<TreeNode>();

                foreach (var node in nodes)
                {
                    if (node.left != null)
                        nextNodes.Add(node.left);

                    if (node.right != null)
                        nextNodes.Add(node.right);
                }

                nodes.Clear();
                nodes.AddRange(nextNodes);

                depth++;
            }
        }

        return depth;
    }
}